bot/exts/evergreen/reddit.py

```diff
@@ -1,128 +1,366 @@
+        self.webhook = None
+        self.access_token = None
+        self.client_auth = BasicAuth(RedditConfig.client_id, RedditConfig.secret)
 
-        Gets a post from r/python by default.
-        Usage:
-        --> .reddit [subreddit_name] [hot/top/new]
-        """
+        bot.loop.create_task(self.init_reddit_ready())
+        self.auto_poster_loop.start()
+
+    def cog_unload(self) -> None:
+        """Stop the loop task and revoke the access token when the cog is unloaded."""
+        self.auto_poster_loop.cancel()
+        if self.access_token and self.access_token.expires_at > datetime.utcnow():
+            asyncio.create_task(self.revoke_access_token())
+
+    async def init_reddit_ready(self) -> None:
+        """Sets the reddit webhook when the cog is loaded."""
+        await self.bot.wait_until_guild_available()
+        if not self.webhook:
+            self.webhook = await self.bot.fetch_webhook(RedditConfig.webhook)
+
+    @property
+    def channel(self) -> TextChannel:
+        """Get the #reddit channel object from the bot's cache."""
+        return self.bot.get_channel(Channels.reddit)
+
+    def build_pagination_pages(self, posts: List[dict], paginate: bool) -> Union[List[tuple], str]:
+        """Build embed pages required for Paginator."""
         pages = []
-        sort_list = ["hot", "new", "top", "rising"]
-        if sort.lower() not in sort_list:
-            await ctx.send(f"Invalid sorting: {sort}\nUsing default sorting: `Hot`")
-            sort = "hot"
+        first_page = ""
+        for post in posts:
+            post_page = ""
+            image_url = ""
 
-        data = await self.fetch(f'https://www.reddit.com/r/{subreddit}/{sort}/.json')
+            data = post["data"]
 
-        try:
-            posts = data["data"]["children"]
-        except KeyError:
-            return await ctx.send('Subreddit not found!')
-        if not posts:
-            return await ctx.send('No posts available!')
+            title = textwrap.shorten(data["title"], width=50, placeholder="...")
+
+            # Normal brackets interfere with Markdown.
+            title = escape_markdown(title).replace("[", "⦋").replace("]", "⦌")
+            link = self.URL + data["permalink"]
+
+            first_page += f"**[{title.replace('*', '')}]({link})**\n"
+
+            text = data["selftext"]
+            if text:
+                first_page += textwrap.shorten(text, width=100, placeholder="...").replace("*", "") + "\n"
+
+            ups = data["ups"]
+            comments = data["num_comments"]
+            author = data["author"]
+
+            content_type = Emojis.reddit_post_text
+            if data["is_video"] is True or {"youtube", "youtu.be"}.issubset(set(data["url"].split("."))):
+                # This means the content type in the post is a video.
+                content_type = f"{Emojis.reddit_post_video}"
+
+            elif any(data["url"].endswith(pic_format) for pic_format in ("jpg", "png", "gif")):
```

Comment:
```suggestion
            elif data["url"].endswith(("jpg", "png", "gif")):
```
`str.endswith` can take a `tuple`.

Reply:
oh thats neat